### ISSUE TITLE
Training a custom OCR

### ISSUE THREAD

**Issue by Jacky2357** [JaidedAI/EasyOCR#1060]
while Training i get training and validation accuracy of about 90% but when i test the custom model my accuracy shows less than 5%.
can u provide step to use custom training .I am using the steps provided in "https://www.youtube.com/watch?v=-j3TbyceShY&t=207s".
Also how can i use trained weights after 20,000 iteration to retrain for another 20000.Means how should i use this or another pretrained  weights  for transfer learning.
The repository doesnot give a clear idea of this.

**Comment by pydoni** [comment]
@Jacky2357 how did you trained? I've tried to use the same tutorial but when I do the error '_MultiProcessingDataLoaderIter' object has no attribute 'next' appears, I've noticed that pytorch change the way you call next, but after changing it starts happening a lenght related error inside the next function

**Comment by akshayrakate** [comment]
For using a pre-trained or last fine-tuned model, add the pre-trained or last fine-tuned model path parameter mentioned in ```en_filtered_config.yaml``` file ```saved_model: '' #'saved_models/en_filtered/iter_300000.pth' ```

1. You can download models using  recognition_models URLs mentioned in the ``easyocr/config.py``` file.
If you want to train gen1 models then change
``` 
optim: adam / adadelta
lr : # Make is small like < 0.001
FeatureExtraction : VGG / ResNet
input_channel: 1
output_channel: 512
hidden_size: 512
new_prediction: False 
freeze_FeatureFxtraction: False  # To freeze VGG/ ResNet weights
freeze_SequenceModeling: False # To freeze BiLSTM weights 
characters: # Mention exact character_list found in recognition_models dictionary
 ``` 
recognition_models dictionary = https://github.com/JaidedAI/EasyOCR/blob/c999505ef6b43be1c4ee36aa04ad979175178352/easyocr/config.py#L53C1-L53C18

**Comment by Gistix** [comment]
> For using a pre-trained or last fine-tuned model, add the pre-trained or last fine-tuned model in `trainer/saved_models` folder and mention model path mentioned in `en_filtered_config.yaml` file to `saved_model: '' #'saved_models/en_filtered/iter_300000.pth' `
> 
> 1. You can **download models using  recognition_models URLs** mentioned in the ``easyocr/config.py``` file.
>    If you want to train gen1 models then **change parameters as per requirements**
> 
> ```
> optim: adam / adadelta
> lr : # Make is small like < 0.001
> FeatureExtraction : VGG / ResNet
> input_channel: 1
> output_channel: 512
> hidden_size: 512
> new_prediction: False 
> freeze_FeatureFxtraction: False  # To freeze VGG/ ResNet weights
> freeze_SequenceModeling: False # To freeze BiLSTM weights 
> characters: # Mention exact character_list found in recognition_models dictionary
> ```
> 
> recognition_models dictionary = https://github.com/JaidedAI/EasyOCR/blob/c999505ef6b43be1c4ee36aa04ad979175178352/easyocr/config.py#L53C1-L53C18

Thank you! Am I able to add new symbols when training from a pre-trained model? The latin charset has ª but not º which we see extensively in our documents.

**Comment by akshayrakate** [comment]
No. You can't add new symbols to the pre-trained model as it will shoot the error as the number of activations in the last layer will be mismatched. 
I am unsure about this but you can try changing the ``` new_prediction: True ``` parameter and adding the new characters to ```characters: ``` in the config file. 
Attaching code link for reference: https://github.com/JaidedAI/EasyOCR/blob/c999505ef6b43be1c4ee36aa04ad979175178352/trainer/train.py#L71C31-L71C31
